### PR TITLE
Began implementing mob damage

### PR DESCRIPTION
**Pull request by ErwinsExpertise**
Implementing mob damage. Still need to work on mob skills, however, this implementation will allow for mob melee damage at a minimum. This code was pulled from commit 4320cdb and adjusted to work with current structure.

### CODE REVIEW

File: server/channel_player.go [hunk 1]
```diff
@@ -648,3 +650,93 @@ func (server ChannelServer) playerMoveInventoryItem(conn mnet.Client, reader mpa
...
+
+		// Magic guard dmg absorption
+
+		// Fighter / Page power guard
+
+		// Meso guard
+
+		plr.GiveHP(int16(playerDamage))
+
+		inst.Send(player.PlayerReceivedDmg(plr.ID(), mobAttack, damage, reducedDamage, spawnID, mobID, healSkillID, stance, reflectAction, reflected, reflectX, reflectY))
```
Comment (Hucaru) [inline server/channel_player.go:736]: Could this be wrapped within the player package? Maybe as `GiveHP(dmg int16, show bool)` (might have situations where negative hp might not need to be shown?) as the player has a ptr to the instance they are in. This also keeps the `PlayerReceiveDmg function` in package scope only and gives it the same nomenclature as the other packet functions.

Comment (ErwinsExpertise) [reply server/channel_player.go:736]: Definitely. I will get that setup and amend the PR. I agree that having it scoped to the player package would be more effective and helpful down the line.

File: server/player/player.go [hunk 2]
```diff
@@ -1058,3 +1058,19 @@ func (d Data) Save(db *sql.DB) error {
...
+	}
+	newHP := d.hp - damage
+
+	if newHP <= -1 {
+		d.hp = 0
+	} else {
+		d.hp = newHP
+	}
+
+	d.Send(packetPlayerStatChange(show, constant.HpID, int32(d.hp)))
```
Comment (Hucaru) [inline server/player/player.go:1075]: The packet function has the bool at the front listed as unknown, Have you tested to see what it does?

Comment (ErwinsExpertise) [reply server/player/player.go]: I did test that, but I did not see any difference for melee or bump damage when it was provided true or false. I had forgotten to remove it from the function. Removed and amended

**Pull request merged**